components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -166,9 +166,22 @@ void JsonRpcService::AddObserver(
 void JsonRpcService::Request(const std::string& json_payload,
                              bool auto_retry_on_network_change,
                              mojom::CoinType coin,
+                             const std::string& chain_id,
                              RequestCallback callback) {
-  RequestInternal(json_payload, auto_retry_on_network_change,
-                  network_urls_[coin], std::move(callback));
```

Comment:
I'm confuse why we need to modify this one and add `RequestSelected` below while we can just get network URL in GetBalance and call RequestInternal directly.

Reply:
Updated to call RequestInternal directly